pkg/component/etcd/statefulset/statefulset.go

```diff
@@ -71,19 +72,49 @@ func (c *component) Deploy(ctx context.Context) error {
 		}
 		sts = c.emptyStatefulset()
 	}
+	numberOfRecreateRequired := c.getNumTimesToRecreateSts(sts)
+	if numberOfRecreateRequired > 0 {
+		return c.recreateStsNTimes(ctx, numberOfRecreateRequired)
+	} else {
+		return c.syncStatefulset(ctx, sts)
+	}
+}
 
+func (c *component) getNumTimesToRecreateSts(sts *appsv1.StatefulSet) uint8 {
 	if sts.Generation > 1 && clusterScaledUpToMultiNode(c.values) && immutableFieldUpdate(sts, c.values) {
-		// Several immutable fields must be reset for the multi-node use-case.
-		deleteAndWait := gardenercomponent.OpDestroyAndWait(c)
-		if err := deleteAndWait.Destroy(ctx); err != nil {
+		return 1
+	} else if c.IsPeerUrlTLSChangedToEnabled() {
+		c.logger.Info("PeerUrl TLS has been enabled for etcd. To reflect this change etcd StatefulSet has to be re-created 2 times", "namespace", c.values.Namespace, "name", c.values.Name, "etcdUID", c.values.EtcdUID)
```

Comment:
Can you add a comment explaining why we need two restarts of all etcd pods?

Reply:
Ok i will add it.